src/settings/window_geometry.rs

```diff
@@ -48,50 +70,49 @@ pub fn try_to_load_last_window_size() -> Result<Dimensions, String> {
 }
 
 pub fn load_last_window_position() -> PhysicalPosition<i32> {
-    let settings_path = neovim_std_datapath();
-    let json = std::fs::read_to_string(&settings_path).map_err(|e| e.to_string());
-    if json.is_err() {
-        return PhysicalPosition::default();
-    }
-    let json = json.unwrap();
-
-    let loaded_position: Result<PhysicalPosition<i32>, _> =
-        serde_json::from_str(&json).map_err(|e| e.to_string());
-    if loaded_position.is_err() {
-        return PhysicalPosition::default();
+    let settings = load_settings();
+    if let Ok(settings) = settings {
+        let loaded_position = settings.window.position;
+        log::debug!("Loaded Window Position: {:?}", loaded_position);
+        loaded_position
+    } else {
+        PhysicalPosition::default()
     }
-    let loaded_position = loaded_position.unwrap();
-    log::debug!("Loaded Window Position: {:?}", loaded_position);
-
-    loaded_position
 }
 
-pub fn maybe_save_window_size(grid_size: Option<Dimensions>) {
+fn maybe_save_window_size(grid_size: Option<Dimensions>) -> Dimensions {
     let settings = SETTINGS.get::<WindowSettings>();
-    let saved_window_size = if settings.remember_window_size {
+    if settings.remember_window_size {
         grid_size.unwrap_or(DEFAULT_WINDOW_GEOMETRY)
     } else {
         DEFAULT_WINDOW_GEOMETRY
```

Comment:
Note, that I retained the old behaviour here of resetting to the default value. But I think it would make more sense to not modify the saved value at all.